src/Functions/decodeXMLComponent.cpp

```diff
@@ -0,0 +1,249 @@
+                        {
+                            char numeric_character = decodeNumberPart(src_curr_pos + 2);
+                            if (numeric_character == '\0')
+                            {
+                                size_t bytes_to_copy = src_next_pos - src_prev_pos;
+                                memcpySmallAllowReadWriteOverflow15(dst_pos, src_prev_pos, bytes_to_copy);
+                                dst_pos += bytes_to_copy;
+                                src_prev_pos = src_curr_pos + 5;
+                            }
+                            else
+                            {
+                                size_t bytes_to_copy = src_curr_pos - src_prev_pos;
+                                memcpySmallAllowReadWriteOverflow15(dst_pos, src_prev_pos, bytes_to_copy);
+                                dst_pos += bytes_to_copy;
+                                *dst_pos = '\0' + numeric_character;
+                                ++dst_pos;
+                                src_prev_pos = src_curr_pos + 5;
+                            }
+                        }
+                        else
+                        {
+                            ++src_curr_pos;
+                            size_t bytes_to_copy = src_curr_pos - src_prev_pos;
+                            memcpySmallAllowReadWriteOverflow15(dst_pos, src_prev_pos, bytes_to_copy);
+                            dst_pos += bytes_to_copy;
+                            src_prev_pos = src_curr_pos;
+                            continue;
+                        }
+                        src_curr_pos += 5;
+                    }
+                    else if (src_next_pos - src_curr_pos == 5)
+                    {
+                        if (strncmp(src_curr_pos, "&quot", 5) == 0)
+                        {
+                            size_t bytes_to_copy = src_curr_pos - src_prev_pos;
+                            memcpySmallAllowReadWriteOverflow15(dst_pos, src_prev_pos, bytes_to_copy);
+                            dst_pos += bytes_to_copy;
+                            *dst_pos = '"';
+                            ++dst_pos;
+                            src_prev_pos = src_curr_pos + 6;
+                        }
+                        else if (strncmp(src_curr_pos, "&apos", 5) == 0)
+                        {
+                            size_t bytes_to_copy = src_curr_pos - src_prev_pos;
+                            memcpySmallAllowReadWriteOverflow15(dst_pos, src_prev_pos, bytes_to_copy);
+                            dst_pos += bytes_to_copy;
+                            *dst_pos = '\'';
+                            ++dst_pos;
+                            src_prev_pos = src_curr_pos + 6;
+                        }
+                        else if (
+                            *(src_curr_pos + 1) == '#' && isdigit(*(src_curr_pos + 2)) && isdigit(*(src_curr_pos + 3))
+                            && isdigit(*(src_curr_pos + 4)))
```

Comment:
The function `isdigit` from libc may depend on locale (it will lead to unexpected results depending on the environment).

We should use a function from `Common/StringUtils.h` instead.